numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -158,12 +160,28 @@ raise_casting_error(
         return -1;
     }
 
+    int j, nin = ufunc->nin, nop = nin + ufunc->nout;
+    PyObject *froms = PyTuple_New(nop);
+    PyObject *tos = PyTuple_New(nop);
```

Comment:
Need to handle the case when these return NULL. Simply giving up and returning the error would be fine.

Reply:
That is:
```C
PyObject *froms = PyTuple_New(nop);
if (froms == NULL) { 
    return NULL;
}
```
etc